widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java

```diff
@@ -0,0 +1,79 @@
+package com.glia.widgets.core.dialog;
+
+import androidx.annotation.NonNull;
+
+import com.glia.widgets.core.dialog.model.DialogState;
+
+import java.util.Deque;
+import java.util.concurrent.LinkedBlockingDeque;
+
+public final class DialogManager {
+    @NonNull
+    private final Callback callback;
+    private final Deque<DialogState> stateDeque;
+    private final DialogState none = new DialogState(Dialog.MODE_NONE);
+
+    public DialogManager(@NonNull Callback callback) {
+        this.callback = callback;
+        stateDeque = new LinkedBlockingDeque<>();
+    }
+
+    public void offer(DialogState state) {
+        if (isDialogShown()) {
+            emitNone();
+        }
+
+        boolean removeLastoccurence = stateDeque.removeLastOccurrence(state);
+        boolean offer = stateDeque.offer(state);
```

Comment:
Looks like `removeLastoccurence` and `offer` variables aren't used, are they?
And minor catch - `removeLastoccurence` => `removeLastOccurence`

Reply:
> Looks like `removeLastoccurence` and `offer` variables aren't used, are they?
> And minor catch - `removeLastoccurence` => `removeLastOccurence`

You're right, I added variables for test purposes and forgot to delete them.